src/software/backend/input/network/networking/network_client.h

```diff
@@ -18,7 +20,7 @@
  * member functions rather than large lambda functions. Overall, this helps keep our
  * main.cpp file shorter and more readable.
  */
-class NetworkClient
+class NetworkClient : public Subject<SensorMsg>
```

Comment:
Why is the `NetworkClient` a subject? Backends output the SensorMsg anyway so don't see why we need to add this here. (Also Observers / Subjects shoudl really only be at the top level so it's very obvious where communication happens)

Reply:
that was what I did to hack the `ReplayLogger` in. will clean up

Reply:
removed